Management/src/ActuatorApi/git.properties

```diff
@@ -0,0 +1,11 @@
+git.remote.origin.url=https://github.com/SteeltoeOSS/Samples
+git.build.version=2.1.0
+git.build.user.email=user@email.com
+git.commit.id.abbrev=90d0870a
+git.commit.id=90d0870a363fafcb50981b7038608b763e527e05
+git.commit.time=2024-10-08T12:30:57-05:00
+git.tags=2.1.0-644-g90d0870a
+git.branch=management
+git.build.time=2024-10-11T13:44:28.9255701-05:00
+git.build.user.name=timhe
```

Comment:
Should this be anonymized?

Reply:
Yeah, that's a better idea... what is here now is just what was generated and I think I had to override the .gitignore for it to be included at all. There are still some oddities with whether or not this file can be generated or included in publishing that I want to better understand... like how can we get the feature to work with `cf push` when the git info isn't included for it to be generated (this isn't working: `<Content Include="git.properties" />`)

Reply:
Integration with msbuild is described at https://docs.steeltoe.io/api/v3/management/info.html:
![image](https://github.com/user-attachments/assets/ad6ea78b-a210-47f7-a041-b345abb287b6)

Reply:
Thanks, I already had that piece included, but the msbuild integration itself does not work when `.git` is not there (such as when source for an app that lives in a monorepo is pushed)

Reply:
Just a wild guess: would it work to include the file from an MSBuild target that runs _after_ one of the [git targets](https://github.com/devlooped/GitInfo/blob/main/src/GitInfo/build/GitInfo.targets), such as [here](https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/openapi/src/Examples/OpenApiKiotaClientExample/OpenApiKiotaClientExample.csproj#L42-L47)?

Reply:
Yeah, there's a condition on the build target so it doesn't fail if the git info isn't available (eg when pushing just the app directory to CF)
```
<Target Name="_GitProperties" AfterTargets="CoreCompile" Condition="!Exists('git.properties')">
```

If you want to see what gets generated, delete the file before build/publish

Reply:
Can't we make it work the other way around, ie generate the file, unless that's not possible like during cf push?

Reply:
After searching for 15 minutes or so, I'm still not sure how we'd do that. There should be a way, but I don't know enough about that project to have an answer

Reply:
I can take a look after my break.

Reply:
I'm attaching a working sample app below that contains a git repository. It can be deployed to CF, but only shows git info at the actuator endpoint when using locally-built binaries (windows or linux). This is because the dotnet-core-buildpack has no access to the .git directory.

Notable changes:
- the project file
- entry in .gitignore
- removed line for email, which isn't available

[GitPropertiesWebApp.zip](https://github.com/user-attachments/files/18154150/GitPropertiesWebApp.zip)

Reply:
The changes from the demo were _not at all_ appropriately taken. Please take time to carefully implement and verify the changes instead of rushing through them without attention. When I build the solution, it checks out the `git.properties` files, which should have been deleted from git in the first place. Comments and sections in project files aren't correct either.

Reply:
The tone of your message is unnecessarily harsh, what I did here was not _that_ far off.

I think `None` is more appropriate than `Content` for the [Build Action](https://learn.microsoft.com/visualstudio/ide/build-actions#build-action-values) since we're not serving `git.properties` as web content.

The `fatal` log entries in CF can only be managed by excluding the package entirely, as will happen with the msbuild conditions I added today. We already discussed my preferences and original intent on the inclusion of the `git.properties` file in source... I think this is in a better state now.

I added a note to the README for more info around `GitInfo`

This screenshot from apps manager shows why email was included in the sample... idk if there is a solution to it though
![image](https://github.com/user-attachments/assets/5527fc4f-075a-40de-948f-4778f2e8f608)

Reply:
> The tone of your message is unnecessarily harsh, what I did here was not _that_ far off.

The applied changes took a different direction than the demo, without any communication about it. Combined with recurring sloppiness elsewhere, I assumed this was another rush. Either way, it didn't work very well. But with the latest changes, things are in good shape.

> I think `None` is more appropriate than `Content` for the [Build Action](https://learn.microsoft.com/visualstudio/ide/build-actions#build-action-values) since we're not serving git.properties as web content.

I agree.

> One thing I don't understand in your demo is why you altered `IncludeAssets`?

Because it's shorter and doesn't seem to make a difference in practice, but I'm fine with keeping the longer version.

> This screenshot from apps manager shows why email was included in the sample... idk if there is a solution to it though

There likely is, but not with the GitInfo package, because it doesn't provide that information. We could switch to something else in the future. There are [more properties](https://github.com/git-commit-id/git-commit-id-plugin-core/blob/master/src/main/java/pl/project13/core/GitCommitPropertyConstant.java) the package doesn't provide (such as the commit message). All the information should be available in the `.git` directory, though. Either way, the email address may be useless because of obfuscation, ie: `xxx@users.noreply.github.com`. The git username would make more sense.